cms/search/publishers.py

```diff
@@ -0,0 +1,202 @@
+    dp-search-data-importer spec link:
+    https://github.com/ONSdigital/dp-search-data-importer/blob/30fb507e90f2cf1974ec0ca43bb0466307e2f112/specification.yml#L186
+    contract: https://github.com/ONSdigital/dis-search-upstream-stub/blob/main/docs/contract/resource_metadata.yml
+    """
+
+    def publish_created_or_updated(self, page: "Page") -> None:
+        """Build the message for the created/updated event.
+        Delegate sending to the subclass's _publish_to_service().
+        """
+        topic = self.get_topic_created_or_updated()
+        message = self._construct_message_for_create_update(page)
+        logger.info(
+            "BasePublisher: About to publish created/updated message=%s to topic=%s",
+            message,
+            topic,
+        )
+        return self._publish_to_service(topic, message)
+
+    def publish_deleted(self, page: "Page") -> None:
+        """Build the message for the deleted event.
+        Delegate sending to the subclass's _publish_to_service().
+        """
+        topic = self.get_topic_deleted()
+        message = self._construct_message_for_delete(page)
+        logger.info(
+            "BasePublisher: About to publish deleted message=%s to topic=%s",
+            message,
+            topic,
+        )
+        return self._publish_to_service(topic, message)
+
+    @abstractmethod
+    def _publish_to_service(self, topic: str, message: dict) -> None:
+        """Each child class defines how to actually send/publish
+        the message (e.g., Kafka, logging, etc.).
+        """
+
+    @abstractmethod
+    def get_topic_created_or_updated(self) -> str:
+        """Provide the topic (or other necessary routing key) for created/updated.
+        This can be a no-op or empty string for some implementations.
+        """
+
+    @abstractmethod
+    def get_topic_deleted(self) -> str:
+        """Provide the topic (or other necessary routing key) for deleted messages.
+        This can be a no-op or empty string for some implementations.
+        """
+
+    def _construct_message_for_create_update(self, page: "Page") -> dict:
+        """Build a dict that matches the agreed metadata schema for 'created/updated'.
+
+        resource_metadata.yml:
+           - StandardPayload: requires {uri, title, content_type}, optional fields
+           - ReleasePayload: extends StandardPayload with release-specific fields
+        """
+        # For pages that are 'release' content, we add release fields below.
+        content_type = self._map_page_type_to_content_type(page)
+
+        # Common fields for StandardPayload (also part of ReleasePayload)
+        # The schema requires at minimum: uri, title, content_type
+        message = {
+            "uri": page.url_path,
+            "content_type": page.content_type_id,
+            "release_date": (
+                page.release_date.isoformat().replace("+00:00", "Z")
```

Comment:
We shouldn't need to change to Zulu; `+00:00` is a valid ISO 8601 format. Was this done for some specific reason?

Reply:
Wanted the time in UTC without explicitly showing the offset, but no other reason, we should be good with `+00:00`.

`Z` Removed!